dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/compilation/VisitorPTX.java

```diff
@@ -33,6 +35,30 @@ public List<Event> visitPtxAtomOp(PTXAtomOp e) {
         );
     }
 
+    // PTX CAS semantics from
+    // https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#parallel-synchronization-and-communication-instructions-atom
+    // atomic {
+    //   d = *a;
+    //   *a = (*a == b) ? c : *a
+    // }
```

Comment:
`d` is not used. Is that a typo or is `d` supposed to be the return value of the CAS (if so, make that more clear)?

Reply:
`d` is the destination/return register. I can add a comment.